src/wallet/wallet2.cpp

```diff
@@ -2555,6 +2530,12 @@ bool wallet2::load_keys(const std::string& keys_file_name, const epee::wipeable_
     const char *field_key_data = json["key_data"].GetString();
     account_data = std::string(field_key_data, field_key_data + json["key_data"].GetStringLength());
 
+    if (json.HasMember("key_on_device"))
```

Comment:
Unneeded, the second to last param in GET_FIELD_FROM_JSON_RETURN_ON_ERROR is "mandatory".